src/ssl/ErrorDetailManager.cc

```diff
@@ -24,6 +26,25 @@ void Ssl::errorDetailClean()
     Ssl::ErrorDetailsManager::Shutdown();
 }
 
+/// ErrorDetailEntry constructor helper that extracts a quoted HTTP field value
+static SBuf
+SlowlyParseQuotedField(const char * const description, const HttpHeader &parser, const char * const fieldName)
+{
+    String fieldValue;
+    if (!parser.hasNamed(fieldName, strlen(fieldName), &fieldValue))
+        throw TextException(ToSBuf("Missing ", description), Here());
```

Comment:
This check is missing in the official code. Without this check httpHeaderParseQuotedString() crashes in the unlikely event an error-details.txt entry is missing a required "detail" or "descr" field. I have disclosed this fix in the updated PR description.

httpHeaderParseQuotedString() also has parsing bugs. I do not think we should touch that old function in this PR. I also do not think we should adjust our SlowlyParseQuotedString() wrapper to compensate for some of those httpHeaderParseQuotedString() problems. The wrapper should remain a wrapper.

Reply:
okay, reverting, adding to my personal TODO list to look into httpHeaderParseQuotedString

Reply:
Okay; I notice that Http::SlowlyParseQuotedString has some checking and uses exceptions for the fail path, so it seems compatible with the intent here. At this point, this wrapper is useless at all, I can just call Http::SlowlyParseQuotedString, as this function only has two callsites.